tests/courseList.spec.ts

```diff
@@ -0,0 +1,96 @@
+import { test, expect } from '@playwright/test'
+
+test.describe.configure({ mode: 'serial' })
+
+test('courselisting works', async ({ page }) => {
+  await page.goto('/archive')
+
+  const heading = page.getByRole('heading', { name: 'Courses' })
+  await expect(heading).toBeVisible()
+})
+
+test('courselisting headers are correct', async ({ page }) => {
+  await page.goto('/archive')
+
+  const headers = page.getByRole('row', { name: 'Course Last modified' })
+  await expect(headers).toBeVisible()
+})
+
+test('courselisting row is correct', async ({ page }) => {
+  await page.goto('/archive')
+
+  const row = page.locator('[data-course-id="2"]')
+  const icon = row.locator('img')
+  const name = row.locator('a', { hasText: 'Ääreisulotteinen lineaarialgebra' })
+  const lastModified = row.locator('time', { hasText: '2018-04-23' })
+
+  await expect(row).toBeVisible()
+  await expect(name).toBeVisible()
+  await expect(lastModified).toBeVisible()
+
+  await expect(icon).toHaveAttribute('src', '/static/img/icon-folder.svg')
+  await expect(name).toHaveAttribute(
+    'href',
+    '/archive/2-aareisulotteinen-lineaarialgebra'
+  )
+})
+
+test('courselisting row hides last modified if no exams', async ({ page }) => {
+  await page.goto('/archive')
+
+  const row = page.locator('[data-course-id="132"]')
+  const lastModified = row.locator('.course-list-item__last-modified')
+  await expect(row).toBeVisible()
+  await expect(row).toHaveText('Probability theory')
+  await expect(lastModified).toBeEmpty()
+})
+
+test('add a new course works', async ({ page, request }) => {
+  await page.goto('/archive')
+
+  const addBox = page.getByText(
+    'Add a new course:Create courseLogged in: dev-user (Log out)'
```

Comment:
Could this be targeted with a test ID? Seems like this may change pretty easily if we change the formatting of the box.

Reply:
Changed to this. +1